### ISSUE TITLE
403 error SignatureDoesNotMatch after a million requests

### ISSUE THREAD

**Issue by rahul003** [aws/aws-sdk-cpp#1072]
### What platform/OS are you using?
Linux/Ubuntu

### What compiler are you using? what version?
gcc 5.4

### What's your CMake arguments?
Built with Bazel
 "PLATFORM_LINUX",
"ENABLE_CURL_CLIENT",
"ENABLE_NO_ENCRYPTION"

https://github.com/tensorflow/tensorflow/blob/master/third_party/aws/BUILD.bazel#L73

### Can you provide a TRACE level log? (sanitize any sensitive information)
Can you please clarify what all is sensitive so I can remove that?
I have the response from S3

```
<Code>SignatureDoesNotMatch</Code>
    <Message>The request signature we calculated does not match the signature you provided. Check your key and signing method.</Message>
    
```

**Comment by wps132230** [comment]
`SinatureDoesNotMatch` may be caused by several reasons, for example clock skew and credentials expiration.
And can you turn on aws log in your application when you are using TensorFlow?

To turn on log, I found some related code here:
https://github.com/tensorflow/tensorflow/blob/4ccb03f468d98d482d6d28108215bb200326b21e/tensorflow/core/platform/s3/s3_file_system.cc#L304
All you need to do is:
```
options.loggingOptions.logLevel = Aws::Utils::Logging::LogLevel::Trace;
```

**Comment by rahul003** [comment]
I do have the trace log for this request. I was not sure what to sanitize, so I didn't post it. Should i sanitize x-amz-security-token along with access key?

**Comment by wps132230** [comment]
Yes, you can replace all these information with * or whatever.

**Comment by wps132230** [comment]
There was a race condition in version 1.3.15, and was fixed in [1.4.54](https://github.com/aws/aws-sdk-cpp/commit/8a7d9f1eace9fee532d4e0186a635c1941d2d5e8)

https://github.com/aws/aws-sdk-cpp/blob/67e1ed5dd2add14359efe6f02097d2597701bd46/aws-cpp-sdk-core/source/auth/AWSAuthSigner.cpp#L454

`m_currentDateStr ` and `m_currentSecretKey ` might be changed by other threads.

**Comment by rahul003** [comment]
Confirming that upgrading did indeed fix it. Thanks!
